src/spinner/src/Spinner.js

```diff
@@ -38,84 +38,69 @@ const innerClass = color =>
     fill: 'transparent'
   }).toString()
 
-class Spinner extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes the Box component as the base.
-     */
-    ...Box.propTypes,
+function Spinner({ delay, size, ...props }) {
+  const theme = useTheme()
+  const [isVisible, setIsVisible] = useState(delay === 0)
+  const [delayTimer, setDelayTimer] = useState(null)
 
-    /**
-     * Delay after which spinner should be visible.
-     */
-    delay: PropTypes.number,
-
-    /**
-     * The size of the spinner.
-     */
-    size: PropTypes.number.isRequired,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
-  }
-
-  static defaultProps = {
-    size: 40,
-    delay: 0
-  }
-
-  constructor({ delay }) {
-    super()
-
-    this.state = {
-      isVisible: delay === 0
+  useEffect(() => {
+    if (delay > 0) {
+      const newDelayTimer = setTimeout(() => {
+        setIsVisible(true)
+      }, delay)
+      setDelayTimer(newDelayTimer)
     }
-  }
 
-  render() {
-    if (!this.state.isVisible) {
-      return null
+    return function() {
+      clearTimeout(delayTimer)
     }
+  }, [])
 
-    const { theme, size, ...props } = this.props
-    return (
-      <Box width={size} height={size} lineHeight={0} {...props}>
-        <Box
-          is="svg"
-          className={outerClass}
-          x="0px"
-          y="0px"
-          viewBox="0 0 150 150"
-        >
-          <Box
-            is="circle"
-            className={innerClass(theme.spinnerColor)}
-            cx="75"
-            cy="75"
-            r="60"
-          />
-        </Box>
-      </Box>
-    )
+  if (!isVisible) {
+    return null
   }
 
-  componentDidMount() {
-    const { delay } = this.props
+  return (
+    <Box width={size} height={size} lineHeight={0} {...props}>
+      <Box
+        is="svg"
+        className={outerClass}
+        x="0px"
+        y="0px"
+        viewBox="0 0 150 150"
+      >
+        <Box
+          is="circle"
+          className={innerClass(theme.spinnerColor)}
+          cx="75"
+          cy="75"
+          r="60"
+        />
+      </Box>
+    </Box>
+  )
+}
 
-    if (delay > 0) {
-      this.delayTimer = setTimeout(() => {
-        this.setState({
-          isVisible: true
-        })
-      }, delay)
-    }
-  }
+Spinner.propTypes = {
+  /**
+   * Composes the Box component as the base.
+   */
+  ...Box.propTypes,
+
+  /**
+   * Delay after which spinner should be visible.
+   */
+  delay: PropTypes.number,
+
+  /**
+   * The size of the spinner.
+   */
+  size: PropTypes.number.isRequired
+}
 
-  componentWillUnmount() {
-    clearTimeout(this.delayTimer)
-  }
+Spinner.defaultProps = {
+  size: 40,
+  delay: 0
 }
 
-export default withTheme(Spinner)
+export default Spinner
```

Comment:
I think to have parity with `PureComponent` we need to support `forwardRef` and `React.memo` right?

Reply:
Cool, not exactly sure what goes into making it work the same as `PureComponent`, I get the `memo` but im not sure how `forwardRef` plays into it. Will take a look tonight!

Reply:
forwardRef will make sure that consumers can still pass `ref` like they could with PureComponent.

We'll have to hook it up to either `innerRef` from ui-box or ref depending on what the inner component is.

Reply:
Solid, thanks for the clarification!